middlewares/userValidations.js

```diff
@@ -0,0 +1,100 @@
+import models from '../models';
+
+const { user } = models;
+/**
+ * @class userValidation
+ */
+class userValidations {
+  /**
+    * Controll update a user.
+    * @constructor
+    * @param {object} req - The request object.
+    * @param {object} res - The response object.
+    * @param {func}  next - The next middleware or function.
+    */
+  static signup(req, res, next) {
+    const usernamecharTest = /[^a-zA-Z/\s/-]/g;
+    const validEmail = /[a-zA-Z0-9]+@[a-z]+\.[com]+/;
+    const usernameTest = /[a-zA-Z]/g;
+    const passwordChar = /[a-zA-Z]/g;
+    const passwordNumb = /[0-9]/g;
+    const passwordTest = /[^\S]/g;
+    let str1, str2, str3;
+
+    if (req.body.email) {
+      user.findOne({ where: { email: req.body.email } })
+        .then((users) => {
+          if (users) {
+            return res.status(400).send(
+              { message: 'Email already in Use, No duplicate emails' });
+          }
+        })
+        .catch(() => res.status(500).json(
```

Comment:
These errors are supposed to be sent to the all catch error handler.

Reply:
ok.